tests/test_modeling_roformer.py

```diff
@@ -0,0 +1,474 @@
+# coding=utf-8
+# Copyright 2021 The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" Testing suite for the PyTorch RoFormer model. """
+
+
+import unittest
+
+from tests.test_modeling_common import floats_tensor
+from transformers import is_torch_available
+from transformers.testing_utils import require_torch, slow, torch_device
+
+from .test_configuration_common import ConfigTester
+from .test_modeling_common import ModelTesterMixin, ids_tensor, random_attention_mask
+
+
+if is_torch_available():
+    import torch
+
+    from transformers import (
+        RoFormerConfig,
+        RoFormerForCausalLM,
+        RoFormerForMaskedLM,
+        RoFormerForMultipleChoice,
+        RoFormerForQuestionAnswering,
+        RoFormerForSequenceClassification,
+        RoFormerForTokenClassification,
+        RoFormerModel,
+    )
+    from transformers.models.roformer.modeling_roformer import ROFORMER_PRETRAINED_MODEL_ARCHIVE_LIST
+
+
+class RoFormerModelTester:
```

Comment:
Could we add one test specifically for the `rotary position embeddings`? C.f.: https://github.com/huggingface/transformers/pull/11684/files?file-filters%5B%5D=.py&file-filters%5B%5D=.rst#r631745463

Think it would be great to have ~2 tests here showcasing how the rotary position embeddings should work